### PR TITLE
LibDNS+LibWeb+Ladybird+RequestServer: Let there be DNS over TLS

### PR DESCRIPTION
**Pull request by alimpfard**

### CODE REVIEW

File: Userland/Utilities/dns.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    MUST(resolver.when_socket_ready()->await());
+
+    size_t pending_requests = requests.size();
+    for (auto& request : requests) {
+        resolver.lookup(request.name, DNS::Messages::Class::IN, request.types)
+            ->when_resolved([&](auto& result) {
+                outln("Resolved {}:", request.name);
+                HashTable<DNS::Messages::ResourceType> types;
+                auto& recs = result->records();
```
Comment (ADKaster) [inline Userland/Utilities/dns.cpp:107]: CI is mad about the auto& here, because `result->records()` returns a copy of the vector of records.

Comment (alimpfard) [reply Userland/Utilities/dns.cpp:107]: Ah, thanks for catching that
I'm surprised it worked on macOS 🤔

Comment (alimpfard) [reply Userland/Utilities/dns.cpp:107]: Oh I think I just tested building ladybird, not the utility

**Review by ADKaster — COMMENTED** [review]
One comment, otherwise lgtm. Beware Tim's "yeet Userland/" PR.

File: Ladybird/HelperProcess.cpp [hunk 2]
```diff
@@ -161,7 +161,19 @@ ErrorOr<NonnullRefPtr<Requests::RequestClient>> launch_request_server_process(Re
         arguments.append(server.value());
     }
 
-    return launch_server_process<Requests::RequestClient>("RequestServer"sv, candidate_request_server_paths, move(arguments));
+    auto client = TRY(launch_server_process<Requests::RequestClient>("RequestServer"sv, candidate_request_server_paths, move(arguments)));
+    WebView::Application::chrome_options().dns_settings.visit(
```
[the remark this answers is not in the document]
Comment (alimpfard) [reply Ladybird/HelperProcess.cpp:165]: I'm a big proponent of not having to restart the application just to change your DNS settings.
We _could_ add it to the command line parameters, but imo this is nicer (just one way to set DNS in RS)

**Review by ADKaster — APPROVED** [review]

**Pull request merged**